### ISSUE TITLE
 High Read Requests Rate for Crossplane Provider Bucket Resource

### ISSUE THREAD

**Issue by abix5** [tagesjump/provider-upjet-yc#17]
Description:
I have noticed a significant increase in the Read Requests rate for the Crossplane Provider Bucket resource after upgrading from version 0.2.0 to version 1.2.0. Previously, the Read Requests rate was stable, but now it has spiked to approximately 40 requests per second.

Metrics:
- ListRequests: N/A
- HeadRequests: 3.66 requests per second
- GetRequests: 36.66 requests per second

Expected Behavior:
The Read Requests rate for the Crossplane Provider Bucket resource should not experience such a significant increase after upgrading to version 1.2.0. The behavior observed in version 0.2.0, where the Read Requests rate was stable, should be maintained.

Steps to Reproduce:
1. Upgrade the Crossplane Provider to version 1.2.0.
2. Create a Bucket resource using the Crossplane Provider.
3. Observe the Read Requests rate for the Bucket resource.

Actual Results:
After upgrading to version 1.2.0, the Read Requests rate for the Bucket resource has increased to approximately 40 requests per second.

Expected Results:
The Read Requests rate for the Bucket resource should remain stable and not experience a significant increase after upgrading to version 1.2.0.

Additional Information:
- Crossplane Provider version: 1.2.0
- Previous Crossplane Provider version: 0.2.0

Please investigate this issue and provide a fix or guidance on how to mitigate the increased Read Requests rate for the Crossplane Provider Bucket resource.

<img width="298" alt="Снимок экрана 2024-08-10 в 18 14 53" src="https://github.com/user-attachments/assets/7ca403de-01d6-4c34-af1b-3e05159e48bc">

**Comment by abix5** [comment]
I have upgraded to version 1.3.0, but unfortunately, it did not resolve the issue. The Read Requests rate for the Crossplane Provider Bucket resource remains high, with no noticeable improvement. Thank you.

**Comment by ssch1337** [comment]
@abix5 Hi, can you tell me at what amount of resources this behavior occurs?

Also, have these crossplane provider configuration settings been changed? (These configurations are set using ControllerConfig or DeployentConfig)

- sync - Sync interval controls how often all resources will be double checked for drift. Default "1h"

- poll - Poll interval controls how often an individual resource should be checked for drift. Default "10m"

- max-reconcile-rate - The global maximum rate per second at which resources may be checked for drift from the desired state. Default "10"

**Comment by abix5** [comment]
All configurations are set to their default values:
sync: 1h
poll: 10m
max-reconcile-rate: 10
We are using the helm repository: https://charts.crossplane.io/stable with version 1.14.5.
As soon as I rolled back to version 0.4.0, everything stabilized. It seems the issue is related to the new architecture introduced from version 1.0.0.
<img width="301" alt="Снимок экрана 2024-08-10 в 19 46 15" src="https://github.com/user-attachments/assets/469ec7db-e4e0-4439-b4cd-8d57731e454e">

**Comment by ssch1337** [comment]
@abix5 And what is the approximate number of Bucket resources you have?

**Comment by abix5** [comment]
the bucket is empty, there are no external requests to them. crossplane creates all these requests

**Comment by ssch1337** [comment]
How many buckets do you have? 1 or more?

**Comment by ssch1337** [comment]
The behavior was reproduced, a fix will be available soon

**Comment by ssch1337** [comment]
Released a new release containing a fix for this bug: [v1.3.1](https://github.com/tagesjump/provider-upjet-yc/releases/tag/v1.3.1)

This occurred because the Yandex provider for the optional `anonymousAccessFlags` field filled it with an array with an empty element during late initialization.

After the update, you will need to either delete the `anonymousAccessFlags` field in the `Bucket` resource using the command below, or fill it in according to the specification
```
kubectl patch buckets.storage.yandex-cloud.upjet.crossplane.io/test-crossplane --type=json -p='[{"op": "remove", "path": "/spec/forProvider/anonymousAccessFlags"}]'
```

Thanks for the detailed description of the issue!
